proto-lens/src/Data/ProtoLens/Encoding.hs

```diff
@@ -47,3 +52,10 @@ parseMessageDelimited = do
     len <- Bytes.getVarInt
     bytes <- Bytes.getBytes $ fromIntegral len
     either fail return $ decodeMessage bytes
+
+-- | Same as @decodeMessage@ but for delimited messages read through a Handle
+decodeMessageDelimitedH :: Message msg => Handle -> ExceptT String IO msg
```

Comment:
Hm, my original suggestion had been to keep ExceptT *internal* to the implementation, and still return `IO (Either String msg)` from this function.

Now that I see this, I'm on the fence which way is better, but leaning towards plain `IO`.  I think `ExceptT` could make sense if we had several different functions we wanted to compose, but I'm not sure we'll add anything beyond this one.

What do you think?  Would `ExceptT` simplify the code you plan to use this with, or would you end up just unwrapping it with `runExceptT` when you used it?

Reply:
I agree with you: returning IO of Either feels more natural in the current interface than returning ExceptT. Let me implement that.

Reply:
Done.